cmd/tailscale/cli/file.go

```diff
@@ -449,9 +443,62 @@ func runFileGet(ctx context.Context, args []string) error {
 		}
 		deleted++
 	}
-	if getArgs.verbose {
+	if deleted == 0 && len(wfs) > 0 {
+		// persistently stuck files are basically an error
+		logf("moved %d/%d files", deleted, len(wfs))
+	} else if getArgs.verbose {
 		printf("moved %d/%d files\n", deleted, len(wfs))
 	}
+	return errs
+}
+
+func runFileGet(ctx context.Context, args []string) error {
+	if len(args) != 1 {
+		return errors.New("usage: file get <target-directory>")
+	}
+	log.SetFlags(0)
+
+	dir := args[0]
+	if dir == "/dev/null" {
+		return wipeInbox(ctx)
+	}
+
+	if fi, err := os.Stat(dir); err != nil || !fi.IsDir() {
+		return fmt.Errorf("%q is not a directory", dir)
+	}
+	nlLogf := logger.Logf(func(format string, args ...any) {
+		printf(format+"\n", args...)
+	})
+	if getArgs.loop {
+		// the rate limiter library effectively uses half of our quota for [RATELIMIT]
+		// messages.  burst has to be at least 2 so we can save up quota for printing
+		// a real message and a summary RATELIMIT message.
+		quietLogf := logger.RateLimitedFn(nlLogf, 15*time.Minute, 4, 125)
+		quietOutln := func(err error) {
+			quietLogf(strings.ReplaceAll(err.Error(), "%", "%%"))
```

Comment:
`quietLogf("%v", err)` instead of shoving it into the format string?

Reply:
That's actually intentional.  We don't want to rate limit all messages together where the format string is "%v" ;-)  Then we'd get a "random sample error message" every half an hour.

It's odd enough that I changed it when you asked and then remembered why I'd done it, and then changed it back, so I've added a  comment.

Reply:
I'm already not a fan of the rate limiting we have in tailscaled. Why's it spreading to the CLI?

Especially with all this mechanism to fight the rate limiter that you added in this PR?

Can we just.... not rate limit here?

You already have the 5 second sleep on failures. (I thought that was going to be an incremental backoff with logtail/backoff.BackOff?). But in any case, what's the point of the quietLogf?

Reply:
Here's the scenario is this: You're running `tailscale file get --skip --loop ~/Downloads` on your linux file server in some systemd user unit to babysit it and capture logs.  Let's call this the "taildrop receiver."  This setup is a bit better than sudo'ing it because as you, it can't be tricked into overwriting system files: it just doesn't have permission.

Ok, so you're browsing on your phone reading papers and slinging the random pdf at your file server.  Then you send some `paper.dvi` at your file server, but there's already a `paper.dvi` in the taildrop directory.  So that file is wedged, but it's one of many you're uploading.  However, every time the taildrop receiver polls tailscaled for new files (every 5 seconds, since there's a persistent error), it logs this same persistent error.  Soon your logs are filling up with this same persistent error.  Every 5 seconds.  Nothing's changed since last time.  It's possible you notice you didn't receive the new `paper.dvi` and look at the logs and see the issue and fix it.  If so great.  You move the other paper to a new name, the taildrop receiver delivers the new file, and the log stops printing out this error message.

If not, in a day you're storing ~1.3MB of logs saying you couldn't receive this `paper.dvi`  With rate limiting to 30 minutes that's down to about ~60KB.  Maybe this is all chump change?  It's duplicated for each file though.  Seems a waste.

Alternates:
1 We could print the error to the log every time, no rate limiting beyond our 5 second persistent error backoff.  It's 1.3MB/day/error.  Acceptable maybe?  Much better than filling the log at line rate.
2  We could save a map of every error we've seen here and print them exactly once.
- Con: there's no way to notice the error "is not happening any more" -- once suppressed, always supressed.
[   Since we're explicitly giving up after 100 failed files in any batch, we've effectively capped the number of log messages we might see if a peer (future feature) starts sending us 2x of a bunch of differently named files, so at least there's not a memory leak attack. ]

Do you have a preference for one of the above two instead of this approach?

The reason I didn't go with your suggestion of using backoff.BackOff for the entire receiver (just the logs) was to keep the flow of other, non-wedged files coming in at a reasonable rate.

Reply:
I'd keep a map keyed off WaitingFile to track its most recent error and when it started failing, logging only on change, and maybe a summary of outstanding errors every N minutes (10, 15, 30? you pick)

I'm waiting for this to grow a Prometheus exporter too so you could plug in monitoring 😂

Reply:
I like it: errors get logged on their rising and falling edges.

 i'm ambivalent to the periodic flushing, but it's in this version.  could easily be ripped out.  also not sure how clear `groundhogsDay` is to the younger generation, so if you or they have better naming ideas i'm all ears.